src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,162 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private T[] array;
+    private int lastElementIndex;
+
+    public ArrayList() {
+        array = (T[])new Object[10];
+        lastElementIndex = -1;
+    }
+
+    public ArrayList(T[] array) {
+        this.array = array;
+        lastElementIndex = array.length - 1;
+    }
+
     @Override
     public void add(T value) {
-
+        if (lastElementIndex < array.length - 1) {
+            ++lastElementIndex;
+            array[lastElementIndex] = value;
+        } else {
+            changeSize(1.5);
```

Comment:
1.5 is a magic number, make it constant
```suggestion
            changeSize(CONSTANT);
```